frontend/src/components/auth/Login.tsx

```diff
@@ -66,14 +88,30 @@ const Login = () => {
     return <Redirect to="/" />;
   }
 
-  if (signup) {
-    return <Redirect to="/signup" />;
-  }
-
   return (
     <div style={{ textAlign: "center" }}>
-      <h1>Login</h1>
+      <h1>{title}</h1>
       <form>
+        {isSignup && (
+          <div>
+            <input
+              type="text"
+              value={firstName}
+              onChange={(event) => setFirstName(event.target.value)}
+              placeholder="First Name"
+            />
+          </div>
+        )}
+        {isSignup && (
```

Comment:
nit: you can wrap the first and last name inputs in the same isSignup check,  ie
```
{isSignup && (
    <div>
    ....... 
    <div>
    .....
)
```